core/domain/exp_services_test.py

```diff
@@ -924,24 +1013,53 @@ def setUp(self):
             'param_changes': []
         }
 
-    def test_update_state_name(self):
-        """Test updating of state name."""
+    def test_state_commands(self):
+        """ Test adding of states."""
         exploration = exp_services.get_exploration_by_id(self.EXP_ID)
+
+        self.assertNotIn('new state', exploration.states)
+
         exp_services.update_exploration(self.owner_id, self.EXP_ID, [{
-            'cmd': 'rename_state',
-            'old_state_name': feconf.DEFAULT_INIT_STATE_NAME,
-            'new_state_name': 'new name',
+            'cmd': exp_domain.CMD_ADD_STATE,
+            'state_name': 'new state',
+        }], 'Add state name')
+
+        exploration = exp_services.get_exploration_by_id(self.EXP_ID)
+        self.assertIn('new state', exploration.states)
+
```

Comment:
FYI: we only use double empty lines when separating classes in a Python file; otherwise we only use single empty lines when spacing out code/class members